version/version_test.go

```diff
@@ -82,3 +85,28 @@ func TestAtLeast(t *testing.T) {
 		})
 	}
 }
+
+func TestVersion_CompareAtPosition(t *testing.T) {
+	tests := []struct {
+		position         VersionPosition
+		currVersion      *Version
+		candidateVersion *Version
+		expected         int
+	}{
+		{Patch, &Version{version: "1.2.0"}, &Version{version: "1.2.3"}, 1},
```

Comment:
Note that we can have versions that include non numeric characters, such as -
```
1.2.3a
````
I believe that the function will return an error in this case. I suggest having the function remove non-numeric characters before initiative the comparison. 

Let's add a test case for such a version as well - 
```
v0.0.2-0.20210315054119-f66993602bf5
```
Let's also have the version not return errors for versions that lookm like -
```
1.0
1
1.4.4-alpha
```
I suggest you look at the code of the existing `Compare` function. It seems to more resilient for such cases.
I think you should also consider using the existing `compareTokens` function, instead of doing it on your own.